CMakeLists.txt

```diff
@@ -11,16 +11,22 @@ enable_testing()
 
 include(cmake/GetGitRevisionDescription.cmake)
 include(cmake/version.cmake)
+include(cmake/CMakeDependentOption.cmake)
 include(GNUInstallDirs)
 include(CheckCXXCompilerFlag)
 
+option(ENABLE_CLANG_JIT "Enable Loading BPF through Clang Frontend" ON)
+option(ENABLE_USDT "Enable User-level Statically Defined Tracing" ON)
+CMAKE_DEPENDENT_OPTION(ENABLE_CPP_API "Enable C++ API" ON "ENABLE_USDT" OFF)
```

Comment:
The idea is mostly to compartmentalize the code, and if this reduces binary size then its a plus. I am aware that the tests fail, for now my goal was just to capture the code dependencies in the library and move the directories around. I don't expect that our buildbots or upstream distros will disable these flags.